library/components/TUrlCopyButton.vue

```diff
@@ -1,57 +1,75 @@
 <template>
-    <div class="relative">
-        
-        <t-tooltip position="bottom" width="130px">
-            <button
-                class="h-[30px] w-[30px] rounded-[4px] flex items-center justify-center border border-[#C3C2CB] p-1 transition-colors"
-                @click="copy"
-                :class="{ 'bg-[#B4E4D9]': copied }"
-                slot="trigger"
-            >
-                <check-icon :size="18" v-if="copied" />
-                <link-icon :size="18" v-else />
-                <span class="visually-hidden">{{ copied ? "URL Copied" : "Copy URL" }}</span>
-            </button>
-            {{ copied ? "URL Copied" : "Copy URL" }}
-        </t-tooltip>
-    </div>
+  <div class="relative">
+    
+    <t-tooltip position="bottom" width="130px">
+      <button
+        class="h-[30px] w-[30px] rounded-[4px] flex items-center justify-center border border-[#C3C2CB] p-1 transition-colors"
+        @click="copy"
+        :class="{ 'bg-[#B4E4D9]': copied }"
+        slot="trigger"
+      >
+        <check-icon :size="18" v-if="copied" />
+        <link-icon :size="18" v-else />
+        <span class="visually-hidden">{{ copied ? "URL Copied" : "Copy URL" }}</span>
+      </button>
+      {{ copied ? "URL Copied" : "Copy URL" }}
+    </t-tooltip>
+  </div>
 </template>
 
 <script>
-    export default {
-        data: () => ({
-            copied: false,
-        }),
-        methods: {
-            copy() {
-                this.copied = true;
-                const tempInput = document.createElement('input'),
-                text = window.location.href;
-                document.body.appendChild(tempInput);
-                tempInput.value = text;
-                tempInput.select();
-                document.execCommand('copy');
-                document.body.removeChild(tempInput);
-                setTimeout(() => {
-                    this.copied = false;
-                }, 1000)
-            }
+  export default {
+    data: () => ({
+      copied: false,
+    }),
+    computed: {
+      url() {
+        const parentUrl = this.filters['context[parent]'];
+
+        if (parentUrl) {
+          const pageContext = 'context[page]=' + this.page.url;
+          return parentUrl + "?" + pageContext + "&" + this.urlParams;
         }
+
+        return window.location.href;
+      },
+      urlParams() {
+        return new URLSearchParams(this.filters).toString();
+      }
```

Comment:
Due to formatting, it might be hard to see changes, I am highlighting the parts that are changed.